### PR TITLE
Add some standard custom configurations and make gzip configurable.

### PR DESCRIPTION
**Pull request by noemi-mancini**

### CODE REVIEW

File: 1.17.6-alpine/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    gzip_comp_level 6;
+    gzip_buffers 16 8k;
+    gzip_http_version 1.1;
+    gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript image/svg+xml;
+
+    ##
+    # Configurations.
+    ##
+    include /etc/nginx/conf.d/*.conf;
+}
```
[the remark this answers is not in the document]
Comment (noemi-mancini) [reply 1.17.6-alpine/nginx.conf]: 1. No, I couldn't make it start on its own without php container, so I copied the default from nginx and then applied what I found in a couple of Zambon projects (but now I managed to do it, so I can replace it if I added too many/wrong configuration)
2. Yes sure

[the remark this answers is not in the document]
Comment (paolomainardi) [reply 1.17.6-alpine/nginx.conf]: 1. The idea is to just rely on the nginx base image:

```
docker create --name nginx nginx:1.17.6-alpine
docker cp nginx:/etc/nginx/nginx.conf .
```

This is the right starting point to apply changes.

[the remark this answers is not in the document]
Comment (noemi-mancini) [reply 1.17.6-alpine/nginx.conf]: @paolomainardi what about these lines?
```    # The default nginx keepalive_timeout is incompatible with the Google Cloud Platform HTTP(S) Load Balancer!
    # Tune nginx keepalives to work with the GCP HTTP(S) Load Balancer:
    keepalive_timeout 620;
    keepalive_requests 10000;
```

[the remark this answers is not in the document]
Comment (paolomainardi) [reply 1.17.6-alpine/nginx.conf]: where this configuration come from ?

File: custom.conf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    server_names_hash_bucket_size 64;
+    map_hash_bucket_size 512;
+    proxy_buffer_size   512k;
+    proxy_buffers   4 512k;
+    proxy_busy_buffers_size   512k;
+
+    ##
+    # Gzip Settings
+    ##
+    gzip on;
```
Comment (paolomainardi) [inline custom.conf]: this one would be nice if we make it configurable, you can see how we hande other confs:

'''
for filename in /etc/nginx/conf.d/*.conf; do
  if [ -e "${filename}" ] ; then
    cp ${filename} ${filename}.tmp
    envsubst '${PHP_HOST} ${PHP_PORT} ${NGINX_DEFAULT_SERVER_NAME} ${NGINX_DEFAULT_ROOT} ${NGINX_SUBFOLDER} ${NGINX_SUBFOLDER_ESCAPED} ${NGINX_OSB_BUCKET} ${NGINX_OSB_RESOLVER} ${DRUPAL_PUBLIC_FILES_PATH} ${NGINX_CACHE_CONTROL_HEADER}' < $filename.tmp > $filename
    rm ${filename}.tmp
  fi
done
```
In this case should be:

```
for filename in /etc/nginx/conf.d//*.conf; do
  if [ -e "${filename}" ] ; then
    cp ${filename} ${filename}.tmp
    envsubst '${PHP_HOST} ${PHP_PORT} ${NGINX_DEFAULT_SERVER_NAME} ${NGINX_DEFAULT_ROOT} ${NGINX_SUBFOLDER} ${NGINX_SUBFOLDER_ESCAPED} ${NGINX_OSB_BUCKET} ${NGINX_OSB_RESOLVER} ${DRUPAL_PUBLIC_FILES_PATH} ${NGINX_CACHE_CONTROL_HEADER}' < $filename.tmp > $filename
    rm ${filename}.tmp
  fi
done
```
We should just add a new env variable 'export NGINX_GZIP_ENABLE=${NGINX_GZIP_ENABLE:-1}` with a default value of 1.

File: 1.17.6-alpine/nginx.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (noemi-mancini) [reply 1.17.6-alpine/nginx.conf]: I found these settings inside the nginx.conf of one project, but maybe it's too related to the infrastructure.

[the remark this answers is not in the document]
Comment (paolomainardi) [reply 1.17.6-alpine/nginx.conf]: we can avoid it for now

**Pull request merged**